tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -302,15 +293,19 @@ void tst_cryptorequests::storedKeyRequests()
     keyTemplate.setFilterData(QLatin1String("test"), QLatin1String("true"));
 
     // first, create the collection via the Secrets API.
-    QDBusPendingReply<Sailfish::Secrets::Result> secretsreply = sm.d_ptr()->createCollection(
-                QLatin1String("tstcryptosecretsgcsked"),
-                DEFAULT_TEST_CRYPTO_STORAGE_PLUGIN_NAME,
-                DEFAULT_TEST_CRYPTO_STORAGE_PLUGIN_NAME,
-                Sailfish::Secrets::SecretManager::DeviceLockKeepUnlocked,
-                Sailfish::Secrets::SecretManager::OwnerOnlyMode);
-    secretsreply.waitForFinished();
-    QVERIFY(secretsreply.isValid());
-    QCOMPARE(secretsreply.argumentAt<0>().code(), Sailfish::Secrets::Result::Succeeded);
+    Sailfish::Secrets::CreateCollectionRequest ccr(&sm);
+    ccr.setCollectionLockType(Sailfish::Secrets::CreateCollectionRequest::DeviceLock);
+    ccr.setCollectionName(QLatin1String("tstcryptosecretsgcsked"));
+    ccr.setStoragePluginName(DEFAULT_TEST_CRYPTO_STORAGE_PLUGIN_NAME);
+    ccr.setEncryptionPluginName(DEFAULT_TEST_CRYPTO_STORAGE_PLUGIN_NAME);
+    ccr.setAuthenticationPluginName(IN_APP_TEST_AUTHENTICATION_PLUGIN);
+    ccr.setDeviceLockUnlockSemantic(Sailfish::Secrets::SecretManager::DeviceLockKeepUnlocked);
+    ccr.setAccessControlMode(Sailfish::Secrets::SecretManager::OwnerOnlyMode);
+    ccr.setUserInteractionMode(Sailfish::Secrets::SecretManager::ApplicationInteraction);
+    ccr.startRequest();
```

Comment:
As a user, I may simply want everything work in blocking mode. Since we don't have such an option, I'd also provide a `performRequest()` method to avoid writing "waiting" macros all the time in the user code.

Reply:
This is true.  Let's do that in a separate PR, as we can add that later in a BC/SC manner, and the other API issues are probably more important to fix first.